RELEASE.md

```diff
@@ -6,27 +6,29 @@ This document simply outlines the release process:
 
 1. Ensure you are running on the oldest supported Erlang version
 
-2. Remove all `-dev` extension from versions (see below for all files)
+1. Remove all `-dev` extension from versions (see below for all files)
```

Comment:
I am not a big fan of the numbering. This document is for the maintainers and I would never render this document as a markdown before reading it. Using `1.` is not helpful to me at all. Can we please keep the old numbers?

Reply:
Okay that's a fair point. I will revert back 👍